sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/sql/CreateTable.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.plans.logical.sql
+
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.catalog.BucketSpec
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.types.StructType
+
+/**
+ * A CREATE TABLE command, as parsed from SQL.
+ *
+ * This is a metadata-only command and is not used to write data to the created table.
+ */
+case class CreateTable(
```

Comment:
Just so I understand correctly, `DDLResolution` returns an instance of `org.apache.spark.sql.execution.datasources.CreateTable`, which is still a subtype of `LogicalPlan`. It's a little confusing at least to me to have two `LogicalPlan` types be named the same thing, but they represent different steps of the query planning, and are in different packages.

What if we called this variant of `CreateTable`, say, making this `ParsedCreateTable` or perhaps `CreateTableStatement`?

Also, when we start partitioning between V1 sources and V2 sources in the Analyzer, are we going to return the same type of `LogicalPlan` back - a `org.apache.spark.sql.execution.datasources.CreateTable` regardless of it's loading V1 or V2? Or should we instead be naming `org.apache.spark.sql.execution.datasources.CreateTableV1`, anticipating the Analyzer to return instances of `CreateTableV2` plans down the road?

Reply:
I'm okay using packages to differentiate between these. The rules use `sql.CreateTable` and I think that is clear.

I'd also like to avoid renaming the v1 classes to include `V1` in their names. That's going to create a lot of conflicts when working with branches and it's hard enough to maintain PRs already given the amount of needless renaming, whitespace changes, and import changes happening.

Similarly, I want to avoid adding `V2` to class names because we eventually want to remove the old packages and support. Then we would be left with a lot of v2 class names, or we would have to refactor to rename. Using packages as qualifiers is well supported in Scala, so I like to use that for this.

Reply:
In the future then when we do split between `v1` and `v2`, perhaps we can clarify by qualifying just the package? Though I guess that goes against Spark convention, but something like this:

```
import org.apache.spark...v1
import org.apache.spark...v2

...

return v1.CreateTable(...)
...
return v2.CreateTable(...)
```
Otherwise we could alias the imports:

```
import org.apache.spark....v1.{CreateTable => CreateTableV1}
import org.apache.spark....v2.{CreateTable => CreateTableV2}
```

Whatever the case then I think disambiguating by package is fine.

Reply:
Sounds good to me! I'll leave this as is for now.

We can also bring this up in the sync to see if there are other opinions.

Reply:
> I'm like using packages to differentiate between the plans.

This reminds me that we add `Exec` postfix to all physical plans, because we think it's not ideal to differentiate plans only be package name. When you search a class by name with an IDE, it will take you more time to pick a class with different package names.

`CreateTableStatemet` sounds good to me.